atc/lockrunner/runner.go

```diff
@@ -19,50 +17,100 @@ type Task interface {
 	Run(context.Context) error
 }
 
+type runner struct {
+	logger           lager.Logger
+	task             Task
+	componentName    string
+	lockFactory      lock.LockFactory
+	componentFactory db.ComponentFactory
+	clock            clock.Clock
+	interval         time.Duration
+}
+
 func NewRunner(
 	logger lager.Logger,
 	task Task,
-	taskName string,
+	componentName string,
 	lockFactory lock.LockFactory,
+	componentFactory db.ComponentFactory,
 	clock clock.Clock,
 	interval time.Duration,
-) ifrit.Runner {
-	return ifrit.RunFunc(func(signals <-chan os.Signal, ready chan<- struct{}) error {
-		close(ready)
-
-		ticker := clock.NewTicker(interval)
-		defer ticker.Stop()
-
-		for {
-			select {
-			case <-ticker.C():
-				lockLogger := logger.Session("tick")
-
-				lock, acquired, err := lockFactory.Acquire(lockLogger, lock.NewTaskLockID(taskName))
-				if err != nil {
-					break
-				}
-
-				if !acquired {
-					lockLogger.Debug(fmt.Sprintln("failed-to-acquire-a-lock-for-", taskName))
-					break
-				}
-
-				ctx := lagerctx.NewContext(context.Background(), lockLogger)
-
-				err = task.Run(ctx)
-				if err != nil {
-					lockLogger.Error("failed-to-run-task", err, lager.Data{"task-name": taskName})
-				}
-
-				err = lock.Release()
-				if err != nil {
-					lockLogger.Error("failed-to-release", err)
-					break
-				}
-			case <-signals:
-				return nil
-			}
+) *runner {
```

Comment:
This seems to have a lot of overlap with the interval runner. Do we need both? Is the only difference that this doesn't use a channel to short-circuit? (No big deal either way I guess.)